subset/switches/src/main/java/switchtest/SwitchTest.java

```diff
@@ -0,0 +1,156 @@
+package switchtest;
+
+import grpc.InterfaceResponse;
+import grpc.LinkStatus;
+import grpc.POENegotiation;
+import grpc.POEStatus;
+import grpc.POESupport;
+import grpc.PowerResponse;
+import grpc.SwitchInfo;
+import grpc.USIServiceGrpc;
+import io.grpc.Channel;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public class SwitchTest {
+
+  private final USIServiceGrpc.USIServiceBlockingStub blockingStub;
+  protected String reportFilename = "tmp/report.txt";
+  protected boolean debug;
+  protected boolean deviceConfigPoeEnabled;
+  protected int rpcTimeoutSec;
+
+  /**
+   * Generic switch test.
+   * @param channel GRPC channel
+   * @param rpcTimeoutSec Timeout in seconds for rpc calls
+   * @param deviceConfigPoeEnabled poe config from module_config
+   * @param debug print debug output
+   */
+  public SwitchTest(Channel channel, int rpcTimeoutSec, boolean deviceConfigPoeEnabled,
+                    boolean debug) {
+    this.debug = debug;
+    blockingStub = USIServiceGrpc.newBlockingStub(channel);
+    this.deviceConfigPoeEnabled = deviceConfigPoeEnabled;
+    this.rpcTimeoutSec = rpcTimeoutSec;
+  }
+
+  protected String testLink(InterfaceResponse interfaceResponse) {
+    if (interfaceResponse.getLinkStatus() == LinkStatus.UP) {
+      return "RESULT pass connection.port_link Link is up\n";
+    }
+    return "RESULT fail connection.port_link Link is down\n";
+  }
+
+  protected String testSpeed(InterfaceResponse interfaceResponse) {
+    int linkSpeed = interfaceResponse.getLinkSpeed();
+    if (linkSpeed > 0) {
+      if (linkSpeed >= 10) {
+        return "RESULT pass connection.port_speed Speed auto-negotiated successfully. "
+            + "Speed is greater than 10 MBPS\n";
+      } else {
+        return "RESULT fail connection.port_speed Speed is too slow. "
+            + "Speed is less than or equal to 10 mbps\n";
+      }
+    }
+    return "RESULT fail connection.port_speed Cannot detect current speed\n";
+  }
+
+  protected String testDuplex(InterfaceResponse interfaceResponse) {
+    String duplex = interfaceResponse.getDuplex();
+    if (duplex != null) {
+      if (duplex.equals("full")) {
+        return "RESULT pass connection.port_duplex Full duplex mode detected\n";
```

Comment:
These strings get ugly after a while... I'd consider extracting to a function or format. E.g. you could have result_string(Boolean pass, String test_name, String detail) and then just say:
return result_string(true, "port_duplex", "Full duplex mode detected")

Generally it's bad form to have a lot of embedded constants, especially a lot of embedded strings.